app/services/LocationService.js

```diff
@@ -55,6 +58,17 @@ export class LocationData {
       let unixtimeUTC = Date.parse(nowUTC);
       let unixtimeUTC_28daysAgo = unixtimeUTC - 60 * 60 * 24 * 1000 * 28;
 
+      // Verify that at least the minimum amount of time between saves has passed
+      // This ensures that no matter how fast GPS coords are delivered, saving
+      // does not happen any faster than the minLocationSaveInterval
+      if (locationArray.length >= 1) {
```

Comment:
I think to make sense use `if(locationArray.length)`, without unnecessary verification